### PR TITLE
Moves autosave functionality to separate file

### LINKED ISSUE #1686: JS: Port over autosave part of patients.coffee from coffeescript to ES6
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

We're porting our custom javascript to ES6! This ticket is to port over:

L57-58 and L65-66 of patients.coffee into app/src/javascript/autosave.js

* **What feature or behavior is this required for?**

Long term project health and maintainability, plus an opportunity to clean up some of our messes when we wrote this the first time.

* **How could we solve this issue? (Not knowing is okay!)**

Rewrite the coffeescript in app/assets/ to be ES6 in app/javascript/src.

* **Anything else?**

See #1679 for an example of this with the `clinic_finders` js.

If this winds up requiring adding dependencies or something, please hit me up in slack!

### PR DESCRIPTION
**Pull request by bravogolfgolf**
I rule and have completed some work on Case Manager that's ready for review!

Moves autosave functionality from patients.coffee to autosave.js

This pull request makes the following changes:
* Removes two function from patients.coffee
* Adds two functions to autosave.js

It relates to the following issue #s: 
* Fixes #1686

### CODE REVIEW

**Comment by colinxfleming** [comment]
the code here looks good, doubly so because this has unearthed an issue with our webpack junk. For whatever reason this isn't autosubmitting, so we should investigate that.

File: app/javascript/src/autosave.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+$(document).on("change", ".edit_patient", () => {
```
Comment (colinxfleming) [inline app/javascript/src/autosave.js]: Can you wrap this whole thing in `$(document).on('turbolinks:load', => {...}`? This is basically the turbolinks version of `$(document).ready` or whatever it is -- we want these handlers to attach to things once we're sure they've loaded.

```suggestion
$(document).on("turbolinks:load", () => {
  $(document).on("change", ".edit_patient", () => {
```

Comment (bravogolfgolf) [reply app/javascript/src/autosave.js]: I think I captured what you are looking for in the latest commit.

**Comment by nikobozi** [comment]
I tried this out locally and apparently the `this` variable is not available within the new arrow function, which is why it's not being executed. After triggering the submit I get the invalid authenticity token error. Maybe it could be something related to [this issue](https://github.com/turbolinks/turbolinks/issues/430#issuecomment-444767978).  
The browser console also keeps complaining about CSP, but I'm not sure what the best option is

**Comment by colinxfleming** [comment]
this has turned out to be a small nightmare. I'm closing this PR for the time being and we should revisit this after we take the steps laid out in #1686 of not double-loading bootstrap.

**Pull request closed without merging**